File: java/leetcode_problems/TwoSum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int num = target-nums[i];
            if(hashMap.containsKey(num)){
                return new int[]{i, hashMap.get(num)};
            }
            hashMap.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: kotlin/leetcode_problems/TwoSum.kt
import java.util.*

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hashMap: HashMap<Int, Int> = HashMap<Int, Int>()
        for(i in 0 .. (nums.size-1)){
            val num: Int = target-nums[i]
            if(hashMap.containsKey(num)){
                return intArrayOf(i, hashMap.get(num)!!)
            }
            hashMap.put(nums[i], i)
        }
        return intArrayOf()
    }
}